### ISSUE TITLE
(question) Customizing components in plain html tags

### ISSUE THREAD

**Issue by artmadeit** [ChristopherBiscardi/gatsby-mdx#285]
Hi, first of all, really love this awesome project!

Is it possible for customize plain html tags?
For example, if I have this..

```markdown
# Hello, world!

## Some h2 in markdown

<h2>Some h2 with html tag</h2>
```
and I have something like this..

```jsx
const MyH2 = ({ children, ...rest}) => <h2 style={{ color: 'tomato' }} {...rest}>{children}</h2>

import { MDXProvider } from "@mdx-js/tag";

export default function Layout({ children }) {
  return (
    <MDXProvider
      components={{
          h2: MyH2,
            }}
    >
      {children}
    </MDXProvider>
  );
}
```

I see only the first `h2` markdown be customized with tomato color (because of `MyH2`) but the plain html tag doesn't. 
In that case what do you recommend, to use in all my `mdx` files just plain markdown elements or set global styles, at least in basic html tags? 

Pd: I'm from peru, sorry for my bad english :stuck_out_tongue:

**Comment by ChristopherBiscardi** [comment]
for now, use the components that you use in the `MDXProvider` in the MDX file as well if you want to have them render the same.

This issue caused me to test something out though, and I have a good idea for how to support replacing the `<h2>Some h2 with html tag</h2>` as well as the `## Some h2 in markdown`. I did a proof of concept yesterday, but I don't know when it will ship.

**Comment by artmadeit** [comment]
Thanks for your advice =) 
Do I close this issue? I think that it could help you as a reminder...

So better i just leave it to you, 
please close this issue when you think is solved.
Again thanks

**Comment by ChristopherBiscardi** [comment]
replacing tags and markdown elements will be possible in MDX 1.0 through `MDXProvider`.

**Comment by ChristopherBiscardi** [comment]
Closing since this was merged for mdx-js 1.0. You can try the `rc` releases of `@mdx-js/mdx`, `@mdx-js/react`, and `gatsby-mdx` for this.
